### PR TITLE
rework on interactions db and impl

### PR DESCRIPTION
**Pull request by rauljareno**

### CODE REVIEW

**Review by plusema86 — APPROVED** [review]

File: lib/iden3comm/domain/use_cases/interaction/add_interaction_use_case.dart [hunk 1]
```diff
@@ -31,21 +36,26 @@ class AddInteractionUseCase
...
-                param: GetIdentityParam(
-                    genesisDid: param.interaction.genesisDid,
-                    privateKey: param.privateKey))
-            .then((_) => _interactionRepository.addInteraction(
-                interaction: param.interaction,
-                genesisDid: param.interaction.genesisDid,
-                privateKey: param.privateKey)))
+    if (param.genesisDid != null &&
+        param.profileNonce != null &&
+        param.privateKey != null) {
```
Comment (Plumillon) [inline lib/iden3comm/domain/use_cases/interaction/add_interaction_use_case.dart:44]: What happens if the integrator just gives one or two params?
Should it throw?

File: lib/iden3comm/domain/use_cases/interaction/get_interactions_use_case.dart [hunk 2]
```diff
@@ -37,20 +40,26 @@ class GetInteractionsUseCase
...
-            .execute(
-                param: GetIdentityParam(
-                    genesisDid: param.genesisDid, privateKey: param.privateKey))
-            .then((_) => _interactionRepository.getInteractions(
-                filters: param.filters,
-                genesisDid: param.genesisDid,
-                privateKey: param.privateKey)))
+    if (param.genesisDid != null &&
+        param.profileNonce != null &&
+        param.privateKey != null) {
```
Comment (Plumillon) [inline lib/iden3comm/domain/use_cases/interaction/get_interactions_use_case.dart:48]: Same comment.

File: lib/iden3comm/domain/use_cases/interaction/remove_interactions_use_case.dart [hunk 3]
```diff
@@ -29,16 +29,18 @@ class RemoveInteractionsUseCase
...
   @override
-  Future<void> execute({required RemoveInteractionsParam param}) {
+  Future<void> execute({required RemoveInteractionsParam param}) async {
     // we check if identity is existing
-    return _getIdentityUseCase
-        .execute(
-            param: GetIdentityParam(
-                genesisDid: param.genesisDid, privateKey: param.privateKey))
-        .then((_) => _interactionRepository.removeInteractions(
+    if (param.genesisDid != null && param.privateKey != null) {
```
Comment (Plumillon) [inline lib/iden3comm/domain/use_cases/interaction/remove_interactions_use_case.dart:34]: Same comment.

File: lib/iden3comm/domain/use_cases/interaction/update_interaction_use_case.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    this._addInteractionUseCase,
+  );
+
+  @override
+  Future<InteractionBaseEntity> execute(
+      {required UpdateInteractionParam param}) async {
+    // we check if profile is valid and identity is existing
+    if (param.genesisDid != null &&
+        param.profileNonce != null &&
+        param.privateKey != null) {
```
Comment (Plumillon) [inline lib/iden3comm/domain/use_cases/interaction/update_interaction_use_case.dart:51]: See above.

**Pull request merged**